### PR TITLE
gcc: make event apply button dynamic

### PR DESCRIPTION
**Pull request by n1tsu**
Replace the 'Apply button' by the newsletter registration widget if there is no incoming event.

### CODE REVIEW

**Comment by n1tsu** [comment]
As mentionned in commit d5ecafd there is major diff that may be caused by a misuse from myself trying to use `manage.py makemessages`.

**Comment by Elfikurr** [comment]
Hello !
Thanks a lot for your work.
I believe however that the 'learn more' page is no longer working !

**Comment by n1tsu** [comment]
> Just to be sure, is the line added to `urls.py` something you really wanted to add ?

If I remember well, this line was added by `black`, but I will see if it is necessary.

> I believe however that the 'learn more' page is no longer working !

Indeed :( Thank you for noticing, I will try to correct it.

File: gccsite/gcc/templates/gcc/registration_widget.html [hunk 1]
```diff
@@ -2,28 +2,34 @@
...
+    </table>
   <p>
     {% if user.is_authenticated %}
       <a href="/application/form/{{last_edition}}" class="btn btn-primary btn-lg btn-block">{% trans 'Apply' %}</a>
     {% else %}
       <a href="{% url 'proloauth_client:autologin' %}?next={{ request.path }}" class="btn btn-primary btn-lg btn-block">{% trans 'Sign in and apply' %}</a>
     {% endif %}
   </p>
+  {% else %}
+    <h3>{% trans "No current summer camps" %}</h3>
```
Comment (fnareoh) [inline gccsite/gcc/templates/gcc/registration_widget.html]: "No current summer camps" sounds a bit strange for me, maybe "No summer camps currently open for registration" would be more accurate? :o

File: gccsite/gcc/views.py [hunk 2]
```diff
@@ -106,8 +106,33 @@ class RessourcesView(TemplateView):
     template_name = "gcc/resources.html"
 
 
-class LearnMoreView(TemplateView):
+class LearnMoreView(FormView):
     template_name = "gcc/learn_more.html"
+    form_class = EmailForm
+    success_url = reverse_lazy("gcc:index")
```
Comment (Elfikurr) [inline gccsite/gcc/views.py:112]: Do we not want to have it go back to /learn ?

**Comment by Elfikurr** [comment]
Since there is email sending that needs to be tested, we will test it once the test environment is ready ! (cc @lportemo )

File: gccsite/gcc/views.py (hunk 2 above)
Comment (n1tsu) [reply gccsite/gcc/views.py:112]: Good point

**Comment by Elfikurr** [comment]
Also, could you change what was said on #78, so that the newsletter widget doesn't appear twice when no event are open for registration ?

**Comment by fnareoh** [comment]
I think we can have it in both cases, as I did it with my last commit it gives the following : 
 no summer camp  some summer camp

**Pull request merged**